src/components/i18n/Param.js

```diff
@@ -0,0 +1,73 @@
+// @flow
+import * as React from 'react';
+
+import isDevEnvironment from '../../utils/env';
+import {
+    JSTYPE_BOOLEAN,
+    JSTYPE_FUNCTION,
+    JSTYPE_NUMBER,
+    JSTYPE_OBJECT,
+    JSTYPE_STRING,
+    JSTYPE_UNDEFINED,
+} from '../../constants';
+
+type Props = {
+    /** The value of this parameter */
+    description: string,
+
+    /** A description of this parameter to help the translators understand what it is */
+    value: any,
+};
+
+/**
+ * @class A placeholder for a replacement parameter in the body of a FormattedCompMessage
+ * component.
+ *
+ * This component renders into the value of the named parameter to the FormattedCompMessage
+ * component. Typically, this component is self-closing.
+ *
+ * @example
+ * <pre>
+ *   <FormattedCompMessage id="x" description="y">
+ *     The file <Param value={filelist[i].path} description="Name of the file that was deleted."/> has been deleted.
+ *   </FormattedCompMessage>
+ * </pre>
+ */
+class Param extends React.Component<Props> {
+    getValue() {
+        const { value } = this.props;
+        switch (typeof value) {
+            default:
+            case JSTYPE_UNDEFINED:
+                return '';
+
+            case JSTYPE_BOOLEAN:
+            case JSTYPE_NUMBER:
+                return String(value);
+
+            case JSTYPE_FUNCTION:
+                return value();
+
+            case JSTYPE_STRING:
+                return value;
+
+            case JSTYPE_OBJECT:
+                if (value === null) {
+                    return '';
+                }
+                if (React.isValidElement(value)) {
+                    return value;
+                }
+                return value.toString();
+        }
+    }
+
+    render() {
+        if (isDevEnvironment() && !this.props.description) {
+            throw new Error('The description property is required on a Param component.');
+        }
+        return this.getValue();
```

Comment:
Minor, but it seems like the logic of `getValue` can live inside `render`.

Reply:
getValue is there for unit testing so we don't have to call render

Reply:
The inclusion of the extra method requires a class, which otherwise isn't necessary. You can use enzyme's `render` helper to test the output of the component's render method.